modules/340-monitoring-custom/templates/service-monitor.yaml

```diff
@@ -17,13 +17,24 @@ spec:
 
   # HTTPS target with samples limited to 1000
   - scheme: https
-    {{- include "tls_config" . | nindent 4 }}
+    {{- include "tls_config" "" | nindent 4 }}
     relabelings:
     {{- include "service_label_selector" . | nindent 4 }}
     {{- include "keep_targets_for_schema" (list "service" "https") | nindent 4 }}
     {{- include "endpoint_by_service_port_name" "https" | nindent 4 }}
     {{- include "base_relabeling" "service" | nindent 4 }}
 
+    {{- if .Values.monitoringCustom.internal.prometheusScraperIstioMTLSEnabled }}
+  # Istio mTLS target with samples limited to 1000
+  - scheme: https
+    {{- include "tls_config" "prometheus-scraper-istio-mtls" | nindent 4 }}
+    relabelings:
+    {{- include "service_label_selector" . | nindent 4 }}
+    {{- include "keep_targets_for_schema" (list "service" "istio-mtls") | nindent 4 }}
+    {{- include "endpoint_by_service_port_name" "http" | nindent 4 }}
```

Comment:
Seems like a problem. In the monitoring-custom module we use port names to distinguish the protocol - `http-metrics` for http and `https-metrics` for https. 

However, this line of the code filters ports by the http-metrics name. Is this the expected behavior? What will happen if users sets the prometheus.deckhouse.io/tls=true annotation?

Reply:
The same question goes to the pod monitor.

Reply:
It is a well-thought approach.
1. Why "http-metrics" — application exports metric in plain http, after enabling istio you don't need to rename ports.
2. We don't support applications that export metric in https. In this case, we need to use two certificates. Probably it worth mentioning in docs.
3. What will happen if users sets the prometheus.deckhouse.io/tls=true annotation? (and istio-mtls=true simultanuesly) — the metrics wont be scraped.